src/OpenLoco/CMakeLists.txt

```diff
@@ -541,10 +541,11 @@ else ()
     endif ()
 endif ()
 
+find_package(span-lite)
```

Comment:
You don't need this find package. It will be taken care of by the other file

Reply:
See above. There's issues with the linker when I remove this. I'll try again later but for now this seems to be needed.

Reply:
that doesn't make any sense. CMake only requires a single find_package to bring the targets into its scope its how the vast majority of projects use this and also how we are using it. To diagnose issues run cmake with [--debug-find](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-debug-find). span-lite produces a config file when it is installed called `span-lite-config.cmake` this has all of the target details for cmake. I'm not sure how flatpak does all its stuff but you'll want to make sure that `--debug-find` is looking at the folder where `span-lite-config.cmake` is output to. It might be an idea to also make this `find_package(span-lite CONFIG)` to force it to not do a module look for span-lite but I doubt anyone has written a `FindSpanLite` file so its unlikely its doing a module look.

Reply:
Good idea. I'll do this later. 

It could be something with the LDFLAGS on my end. We're nearly there!

Reply:
span-lite is header only so shouldn't have anything to do with LDFLAGS.

Reply:
```
find_package considered the following locations for span-lite's Config
  module:

    /run/build/OpenLoco/_flatpak_build/CMakeFiles/pkgRedirects/span-liteConfig.cmake
    /run/build/OpenLoco/_flatpak_build/CMakeFiles/pkgRedirects/span-lite-config.cmake
    /usr/lib/sdk/toolchain-i386/span-liteConfig.cmake
    /usr/lib/sdk/toolchain-i386/span-lite-config.cmake
    /app/span-liteConfig.cmake
    /app/span-lite-config.cmake
    /app/lib32/cmake/span-lite/span-liteConfig.cmake
    /app/lib32/cmake/span-lite/span-lite-config.cmake

  The file was found at

    /app/lib32/cmake/span-lite/span-lite-config.cmake



-- Configuring done
CMake Error at src/OpenLoco/CMakeLists.txt:544 (target_link_libraries):
  Target "OpenLoco" links to:

    nonstd::span-lite

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
  ```

So - the package is found but it has an issue with the namespace.

Unless I add find_package in the src/OpenLoco CMake file for some reason and I've got no idea as to why. Will keep trying some things but it is pretty confusing.

Reply:
Can you provide the info from the config file

Reply:
It should also have a target file with it

Reply:
Thanks @duncanspumpkin - 
[span-lite-config.txt](https://github.com/OpenLoco/OpenLoco/files/10098577/span-lite-config.txt)
[span-lite-targets.txt](https://github.com/OpenLoco/OpenLoco/files/10098595/span-lite-targets.txt)

I wonder if it is a timing thing, or if the fetch content plugin does something else to the process. Does the fetch content plugin simply just pull the code into the folder?

Reply:
Hey - disregard this. It works with the latest changes on master.

Reply:
Hmm okay so that is weird. The config is looking for `span-lite::span-lite` the targets `nonstd::span-lite`

Reply:
Argh its a mistake in span-lite. I'll send them a pr.

Reply:
Cheers for your help!

 It's super weird that it works when I add a second find_package call in the src/OpenLoco/CMakeLists.txt file.

Reply:
Tried the merged changes from the span-lite repo from the Flatpak build, it still errors out in the build. I guess I just need to work out what's happening with the targets.

Reply:
Unfortunately i realised this wasn't the issue after sending my pr. What version of cmake is being used? If it's 3.24 you could try add `GLOBAL` to the find package call.

Reply:
I'm just reading more into this and it sounds like the find package scope is the issue. We can get around this by using `include` instead of `add_directory` for the third-party cmake file. That will bring the targets scope into the parent and will be available for the children.

Reply:
Thanks for this - I'll try it later on my branch.

Reply:
Thanks for the suggestion here. GLOBAL did the trick. I've removed the extra find_package as a result.

Reply:
Global would work but I'm not sure if all our ci's have new enough CMake for that.

Reply:
Yes I see this. 

Are these based on the docker images? If so could we simply upgrade the version of cmake in the containers?

Reply:
Deleted previous comment. I've managed to do it with import instead of add_subdirectory. The reason it didn't work before was because I needed to change the order to pull in thirdparty before adding src.